zerver/models/users.py

```diff
@@ -78,6 +78,21 @@ class UserBaseSettings(models.Model):
     COLOR_SCHEME_CHOICES = [COLOR_SCHEME_AUTOMATIC, COLOR_SCHEME_NIGHT, COLOR_SCHEME_LIGHT]
     color_scheme = models.PositiveSmallIntegerField(default=COLOR_SCHEME_AUTOMATIC)
 
+    # Information density is established user by user through
+    # adjustments to the font size, line height, and full line
+    # space (the amount of space between one block element and
+    # another).
+    WEB_FONT_SIZE_PX_LEGACY = 14
+    WEB_LINE_HEIGHT_PX_LEGACY = 20
+    # Default to 100% of the line height for full-line space.
+    # This setting allows, for example, users to prefer a
+    # tighter line-height, but with more generous, skimming-
+    # friendly spaces between block elements in messages.
+    WEB_FULL_LINE_SPACE_PX = WEB_LINE_HEIGHT_PX_LEGACY
+    web_font_size_px = models.PositiveSmallIntegerField(default=WEB_FONT_SIZE_PX_LEGACY)
+    web_line_height_px = models.PositiveSmallIntegerField(default=WEB_LINE_HEIGHT_PX_LEGACY)
+    web_full_line_space_px = models.PositiveSmallIntegerField(default=WEB_FULL_LINE_SPACE_PX)
```

Comment:
I'm reasonably happy with this structure at a proposal; I started [an API design thread](https://chat.zulip.org/#narrow/stream/378-api-design/topic/web.20information.20density.20settings/near/1753264) to get approval for merging this.